core/packages/contracts/src/OutboundChannel.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/AccessControl.sol";
+import "./IOutboundChannel.sol";
+import "./IVault.sol";
+
+contract OutboundChannel is IOutboundChannel, AccessControl {
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant SUBMIT_ROLE = keccak256("SUBMIT_ROLE");
+
+    mapping(bytes => uint64) public nonce;
+
+    IVault public vault;
+    uint256 public fee;
+
+    event Message(bytes dest, uint64 nonce, bytes payload);
+
+    event FeeUpdated(uint256 fee);
+
+    error FeePaymentToLow();
+
+    constructor(IVault _vault, uint256 _fee) {
+        _grantRole(ADMIN_ROLE, msg.sender);
+        _setRoleAdmin(SUBMIT_ROLE, ADMIN_ROLE);
```

Comment:
need admin revoke?

also, can't find where new SUBMIT_ROLE contracts are added? doesn't seem to be in Deploy? where does the NativeTokens get submit permission?